main.tf

```diff
@@ -103,102 +105,71 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
+
     content {
       allowed_methods  = default_cache_behavior.value.allowed_methods
       cached_methods   = default_cache_behavior.value.cached_methods
       target_origin_id = default_cache_behavior.value.target_origin_id
-      compress         = default_cache_behavior.value.compress
+      compress         = lookup(default_cache_behavior.value, "compress", null)
 
       forwarded_values {
         query_string = default_cache_behavior.value.query_string
         cookies {
           forward = default_cache_behavior.value.cookies_forward
         }
-        headers = default_cache_behavior.value.headers
+        headers = lookup(default_cache_behavior.value, "headers", null)
       }
 
       dynamic "lambda_function_association" {
-        for_each = default_cache_behavior.value.lambda_function_association == null ? [] : [for i in default_cache_behavior.value.lambda_function_association : {
-          event_type   = i.event_type
-          lambda_arn   = i.lambda_arn
-          include_body = i.include_body
-        }]
+        iterator = lambda
+        for_each = lookup(default_cache_behavior.value, "lambda_function_association", [])
         content {
-          event_type   = lambda_function_association.value.event_type
-          lambda_arn   = lambda_function_association.value.lambda_arn
-          include_body = lambda_function_association.value.include_body
+          event_type   = lambda.value.event_type
+          lambda_arn   = lambda.value.lambda_arn
+          include_body = lookup(lambda.value, "include_body", null)
         }
       }
 
       viewer_protocol_policy = default_cache_behavior.value.viewer_protocol_policy
-      min_ttl                = default_cache_behavior.value.min_ttl
-      default_ttl            = default_cache_behavior.value.default_ttl
-      max_ttl                = default_cache_behavior.value.max_ttl
+      min_ttl                = lookup(default_cache_behavior.value, "min_ttl", null)
+      default_ttl            = lookup(default_cache_behavior.value, "default_ttl", null)
+      max_ttl                = lookup(default_cache_behavior.value, "max_ttl", null)
     }
   }
 
   dynamic "ordered_cache_behavior" {
-    for_each = [for i in var.dynamic_ordered_cache_behavior : {
-      path_pattern                = i.path_pattern
-      allowed_methods             = i.allowed_methods
-      cached_methods              = i.cached_methods
-      target_origin_id            = i.target_origin_id
-      compress                    = i.compress
-      query_string                = i.query_string
-      cookies_forward             = i.cookies_forward
-      headers                     = i.headers
-      viewer_protocol_policy      = i.viewer_protocol_policy
-      min_ttl                     = i.min_ttl
-      default_ttl                 = i.default_ttl
-      max_ttl                     = i.max_ttl
-      lambda_function_association = lookup(i, "lambda_function_association", null)
-    }]
```

Comment:
I am not sure this can be removed since you will potentially need to iterate over multiple ordered behaviors

Reply:
it can be removed 
haven't yet tested this one will do soon this is the recommended syntax

Reply:
tested locally, confirming iterating works with this syntax